enterprise/frontend/src/embedding-sdk/components/public/StaticQuestion/StaticQuestion.unit.spec.tsx

```diff
@@ -174,4 +175,15 @@ describe("StaticQuestion", () => {
       value: 1024,
     });
   });
+
+  it("should cancel the request when the component unmounts", async () => {
+    const abortSpy = jest.spyOn(AbortController.prototype, "abort");
```

Comment:
The `Api` class in `metabase/lib/api.js` calls `abortController.abort()` when the request is cancelled. I tried using `fetchMock` for this, but that does not work as the fetchMock request immediately fulfills a request on render.

Reply:
This doesn't check if that specific request has been aborted though right?
Meaning any other request being aborted would make the test pass

Reply:
The requests are immediately fulfilled because we have mocks in place that return a static response in the setup function, I tried to comment those lines and re-mocking the request putting a sleep in the response, but it seems like fetch mock is not giving us access to the abort signal :(
Maybe you have better ideas:
```
 it.only("should cancel the request when the component unmounts", async () => {
    const abortControllers: AbortController[] = [];

    fetchMock.get({ url: `path:/api/card/1` }, async (url, opts) => {
      const controller = new AbortController();
      abortControllers.push(controller);
      console.log({ opts });
      opts.signal = controller.signal;
      await sleep(1000);
      return "hello";
    });

    fetchMock.post({ url: `path:/api/card/1/query` }, async (url, opts) => {
      const controller = new AbortController();
      abortControllers.push(controller);
      opts.signal = controller.signal;
      await sleep(10000);
      return "hello";
    });

    const { unmount } = setup();

    await sleep(500);

    await act(async () => unmount());

    await sleep(500);

    const lastGet = fetchMock.lastCall(`path:/api/card/1`);
    const lastPost = fetchMock.lastCall(`path:/api/card/1/query`);

    console.log({ lastGet, lastPost });

    for (const controller of abortControllers) {
      // doesnt seem to be working
      expect(controller.signal.aborted).toBe(true);
    }

    // also doesn't work
    expect(lastGet?.[1].signal?.aborted).toBe(true);
    expect(lastPost?.[1].signal?.aborted).toBe(true);
  });
```

Reply:
If the number of cancelled requests is stable, we could check for that otherwise, or at least check that it's been called at least twice

Reply:
@npretto I tried using the built-in delay feature of `fetchMock` but I wasn't able to mock the abort signal yesterday as well 😢, let me assert the number of requests/calls 👍🏻

Reply:
It's a bit sad that some things are so hard to test :/

Reply:
I added a [simple sanity check](https://github.com/metabase/metabase/pull/48808/commits/f7851b9d95583845ba9e68084e09e0ec217ca454) as fetch-mock doesn't know if a particular request is cancelled or not 🫠

Reply:
Shouldn't we have checked if abort had been called twice?

Reply:
@npretto Oh, I misunderstood that a bit, that makes sense - I'll update the test in a follow-up PR then